src/gui/progressbardelegate.cpp

```diff
@@ -28,47 +28,41 @@
 
 #include "progressbardelegate.h"
 
-#include <QApplication>
-#include <QModelIndex>
 #include <QPainter>
+#include <QPalette>
 #include <QStyleOptionViewItem>
 
 #if (defined(Q_OS_WIN) || defined(Q_OS_MACOS))
 #include <QProxyStyle>
 #endif
 
-ProgressBarDelegate::ProgressBarDelegate(const int progressColumn, const int dataRole, QObject *parent)
-    : QStyledItemDelegate {parent}
-    , m_progressColumn {progressColumn}
-    , m_dataRole {dataRole}
+ProgressBarDelegate::ProgressBarDelegate()
 {
 #if (defined(Q_OS_WIN) || defined(Q_OS_MACOS))
-    m_dummyProgressBar.setStyle(new QProxyStyle {"fusion"});
```

Comment:
there was a memory leak here 🙄 
https://doc.qt.io/qt-5/qwidget.html#setStyle:
>The ownership of the style object is not transferred.

Reply:
The lifetime of ProgressBarDelegate in all cases is the application's lifetime. so I will say it's a "soft" leak.

Reply:
Still bad anyway.

Reply:
>The lifetime of ProgressBarDelegate in all cases is the application's lifetime. 

You shouldn't rely on this inside ProgressBarDelegate itself.